imported from OmniOS
Let gold use the correct ELF interpreter

--- binutils-2.43.1/gold/x86_64.cc.orig
+++ binutils-2.43.1/gold/x86_64.cc
@@ -1301,7 +1301,11 @@
   true,			// is_default_stack_executable
   true,			// can_icf_inline_merge_sections
   '\0',			// wrap_char
+#if defined(__illumos__)
+  "/lib/64/ld.so.1",
+#else
   "/lib/ld64.so.1",     // program interpreter
+#endif
   0x400000,		// default_text_segment_address
   0x1000,		// abi_pagesize (overridable by -z max-page-size)
   0x1000,		// common_pagesize (overridable by -z common-page-size)
